### PR TITLE
Reenable possibility to use an already installed C-Blosc library

### LINKED ISSUE #216: BLOSC_DIR environment variable seems defunct as of v1.8.3
The `BLOSC_DIR` build environment variable does not seem to be used anywhere.
A search on the repository only shows https://github.com/Blosc/python-blosc/search?q=BLOSC_DIR&unscoped_q=BLOSC_DIR :
- https://github.com/Blosc/python-blosc/blob/v1.8.3/doc/install.rst#compiling-with-an-installed-blosc-library
- https://github.com/Blosc/python-blosc/blob/v1.8.3/README.rst#compiling-with-an-installed-blosc-library
- https://github.com/Blosc/python-blosc/blob/v1.8.3/setup.py#L97

The `install.rst` and `README.rst` also have to be amended to remove the `--blosc=/usr/local` option.

### PR DESCRIPTION
**Pull request by bnavigator**
This fixes #216, by introducing a `USE_SYSTEM_BLOSC` environment variable. Custom install locations are supported through CMake's standard settable `Blosc_ROOT`,  `Blosc_INCLUDE_DIR`, etc. variables for [finding modules](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules).

For consistency with the standard CMake way, I decided against retaining the defunct `BLOSC_DIR` variable.

BTW, the `build_clib` and `build_ext` commands seems to do nothing on my installs, everything is performed by scikit-build during the `build` stage. Might have the same reason as #222.

### CODE REVIEW

**Comment by sreerajkksd** [comment]
@FrancescAlted @thewtex  Can one of you review this PR ? Does this approach look sane to you guys ?

**Comment by thewtex** [comment]
Hi @bnavigator @sreerajkksd @FrancescAlted I explained more in #216, but improvements could be made by using a CMake package configuration file.

**Comment by FrancescAlted** [comment]
Any possibility that the PR adopts the suggestions by @thewtex in #212?  Also, the PR has conflicts and needs an update.

**Comment by bnavigator** [comment]
If I understand correctly, @thewtex suggests to create a CMake package file instead of this PRs FindBlosc.cmake. This would go into c-blosc, not python-blosc. Way out of scope of this PR.

File: README.rst
Comment (FrancescAlted) [inline README.rst:119]: This mix of upper and lower case in `Blosc_ROOT` looks a bit inconsistent.  I suggest to use `BLOSC_ROOT`.

File: cmake/FindBlosc.cmake
Comment (FrancescAlted) [inline cmake/FindBlosc.cmake:9]: Ditto for `Blosc_FOUND`, `Blosc_INCLUDE_DIRS`, `Blosc_LIBRARIES`, `Blosc_INCLUDE_DIR` and `Blosc_LIBRARY`.

**Review by FrancescAlted — CHANGES_REQUESTED** [review]
This looks reasonable to me (unless @thewtex would propose otherwise).

File: README.rst
Comment (bnavigator) [reply README.rst:119]: Please consult the CMake documentation and reconsider: https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#standard-variable-names

Comment (FrancescAlted) [reply README.rst:119]: Ok.  I can see that CMake guys prefer the mixed case, that's ok.  However, I'd continue advocating for making the *environment* variable consistently all in upper case.  If you want, you can create a new variable inside CMake that is mixed case.  Sorry if I seem a bit picky about this, but I think taking this additional action would save some headaches to users.

File: cmake/FindBlosc.cmake
Comment (FrancescAlted) [reply cmake/FindBlosc.cmake:9]: Ok.  That's inside CMake config files, so no problem with me.

File: README.rst
Comment (bnavigator) [reply README.rst:119]: [The environment variable name is a direct consequence of the name of FindBlosc.cmake due to how CMake handles find_modules](https://cmake.org/cmake/help/latest/envvar/PackageName_ROOT.html#envvar:%3CPackageName%3E_ROOT). This line is just a hint in the README how to manipulate CMake for non-standard setups.

Comment (FrancescAlted) [reply README.rst:119]: Holly cow.  I definitely disagree with conventions that CMake team is making even for *environment* variables.  But if that is the naming hell where we are all headed, be it.

**Review by FrancescAlted — COMMENTED** [review]
Let's keep this on-hold until/if @thewtex would chime in.

File: README.rst
Comment (bnavigator) [reply README.rst:119]: I think you underestimate the power of how CMake handles it's variables. Envvars, cached vars, command-line flags, they are all related somehow and have a predetermined precedence.

File: cmake/FindBlosc.cmake
Comment (FrancescAlted) [inline cmake/FindBlosc.cmake:12]: Add a new line at the end of the file.

**Review by FrancescAlted — COMMENTED** [review]
I have just a trivial comment

File: cmake/FindBlosc.cmake
Comment (bnavigator) [inline cmake/FindBlosc.cmake:12]:
```suggestion
endif ()

```

File: README.rst
Comment (FrancescAlted) [reply README.rst:119]: Yeah, maybe.  I just don't like users being presented with these conventions.  As said, they look weird to my eyes (but hopefully it is only me :-P).

**Review by thewtex — APPROVED** [review]

**Comment by thewtex** [comment]
> @thewtex suggests to create a CMake package file instead of this PRs FindBlosc.cmake

Yes, we should definitely be using a CMake package configuration file instead of a Find module. But, if @bnavigator is not willing to make those changes, this is still a step forward. We could merge this and make additional progress later.

**Comment by sreerajkksd** [comment]
I would recommend merging this now (as we have approval from the maintainers) and creating a new PR for the concern mentioned. I can see that @bnavigator already created an issue for this. I believe this is going to help some people (including me) to compile with an existing c-blosc library.

**Comment by sreerajkksd** [comment]
@FrancescAlted Can you cut a new release in Pypi ?

**Comment by eamanu** [comment]
HI, 

I'm packing python-blosc 1.10.6 in Debian. I wanted  to be sure if this PR won't re
introduce the issue https://github.com/Blosc/python-blosc/issues/227 and fixed here https://github.com/Blosc/python-blosc/pull/242/files.

Thanks!
Cheers,

**Pull request merged**
